do-not-commit/list-modify-iterating.py

```diff
@@ -0,0 +1,32 @@
+l = list(range(100))
+# ruleid:list-modify-while-iterate
+for i in l:
+    print(i),
+    print(l.pop(0))
+    x = l.pop(0)
+    print(x)
+
+a = [1, 2, 3, 4]
+# ruleid:list-modify-while-iterate
+for i in a:
+    print(i)
+    a.pop(0)
+
+b = [1, 2, 3, 4]
+# ruleid:list-modify-while-iterate
+for i in b:
```

Comment:
It appears that `b` is a list that is being modified while in a for loop.
This will likely cause a runtime error or an infinite loop. ([.list-modify-iterating.list-modify-while-iterate](https://semgrep.dev/r/.list-modify-iterating.list-modify-while-iterate))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.list-modify-iterating.list-modify-while-iterate) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>